pkg/hubble/exporter/config_watcher.go

```diff
@@ -67,7 +67,7 @@ func (c *configWatcher) reload() {
 	config, hash, err := c.readConfig()
 	if err != nil {
 		DynamicExporterReconfigurations.WithLabelValues("failure").Inc()
-		c.logger.Warnf("failed reading dynamic exporter config")
+		c.logger.Warnf("failed reading dynamic exporter config: %s", err)
```

Comment:
👍 

should [this](https://github.com/cilium/cilium/blob/v1.16.2/pkg/hubble/exporter/config_watcher.go#L131-L133) not also be updated to allow multiple export configs that use `stdout` as `filePath`?

```go
if _, ok := flowlogPaths[filePath]; ok {
	 errs = errors.Join(errs, fmt.Errorf("duplicated flowlog path %s", filePath))
}
```

e.g.

```yaml
hubble:
  export:
    dynamic:
    static:
      enabled: true
      config:
        content:
        - name: flow-logs
          filePath: stdout
          fieldMask:
          - time
          - source.namespace
          - source.pod_name
          - IP.source
          - destination.namespace
          - destination.pod_name
          - IP.destination
          - l4
          - verdict
          includeFilters:
          - source_label:
            - k8s:networking.everquote.com/flow-logs=enabled
            event_type:
            - type: 1
            - type: 4
          - destination_label:
            - k8s:networking.everquote.com/flow-logs=enabled
            event_type:
            - type: 1
            - type: 4
          excludeFilters:
          - source_label:
            - k8s:networking.everquote.com/flow-logs=detailed
          - destination_label:
            - k8s:networking.everquote.com/flow-logs=detailed
        - name: flow-logs-detailed
          filePath: stdout
          fieldMask: []
          includeFilters:
          - source_label:
            - k8s:networking.everquote.com/flow-logs=detailed
          - destination_label:
            - k8s:networking.everquote.com/flow-logs=detailed
          excludeFilters: []
```

Reply:
> 👍
> 
> should [this](https://github.com/cilium/cilium/blob/v1.16.2/pkg/hubble/exporter/config_watcher.go#L131-L133) not also be updated to allow multiple export configs that use `stdout` as `filePath`?
> 
> ```go
> if _, ok := flowlogPaths[filePath]; ok {
> 	 errs = errors.Join(errs, fmt.Errorf("duplicated flowlog path %s", filePath))
> }
> ```

Good point, let me take a closer look.

Reply:
Unfortunately, I don't think that check can be simply removed, even for stdout without more consideration, and likely some more advanced configuration because each `FlowLogConfig` represents 1 exporter. Every flow is sent to every exporter, meaning if we allowed you to specify `stdout` more than once, you would see some flows logged multiple times.